packages/core/src/repository/cached-repository.provider.ts

```diff
@@ -0,0 +1,103 @@
+import { RepositoryProvider, RepositoryInfo, LicenseInfo, ContributingInfo, RepositoryOptions } from './repository.provider.js';
+import { Repository } from './repository.service.js';
+
+/**
+ * Abstract base class that provides caching for repository providers
+ * Subclasses should implement the fetch* methods to retrieve data
+ */
+export abstract class CachedRepositoryProvider<Options extends RepositoryOptions = RepositoryOptions> implements RepositoryProvider<Options> {
+  private cache = new Map<string, unknown>();
+
+  abstract getPlatformName(): string;
+  abstract getPriority(): number;
+  abstract supports(): Promise<boolean>;
+  abstract getOptions(): import('./repository.provider.js').RepositoryOptionsDescriptors<Options>;
+  abstract setOptions(options: Partial<Options>): void;
+
+  /**
+   * Fetch basic repository information (to be implemented by subclasses)
+   */
+  protected abstract fetchRepositoryInfo(): Promise<RepositoryInfo>;
+
+  /**
+   * Fetch repository logo URI (to be implemented by subclasses)
+   */
+  protected abstract fetchLogo(): Promise<string | undefined>;
+
+  /**
+   * Fetch license information (to be implemented by subclasses)
+   */
+  protected abstract fetchLicense(): Promise<LicenseInfo | undefined>;
+
+  /**
+   * Fetch contributing guidelines information (to be implemented by subclasses)
+   */
+  protected abstract fetchContributing(): Promise<ContributingInfo | undefined>;
+
+  /**
+   * Get basic repository information with caching
+   */
+  async getRepositoryInfo(): Promise<RepositoryInfo> {
+    return this.getCached('repositoryInfo', () => this.fetchRepositoryInfo());
+  }
+
+  /**
+   * Get repository logo URI with caching
+   */
+  async getLogo(): Promise<string | undefined> {
+    return this.getCached('logo', () => this.fetchLogo());
+  }
+
+  /**
+   * Get license information with caching
+   */
+  async getLicense(): Promise<LicenseInfo | undefined> {
+    return this.getCached('license', () => this.fetchLicense());
+  }
+
+  /**
+   * Get contributing guidelines information with caching
+   */
+  async getContributing(): Promise<ContributingInfo | undefined> {
+    return this.getCached('contributing', () => this.fetchContributing());
+  }
+
+  /**
+   * Get complete repository information (convenience method)
+   */
+  async getRepository(): Promise<Repository> {
```

Comment:
We don't want to use getRepository anymore, only specific data retrieval methods